tests/integration/java-integration/src/test/java/io/fluentbit/test/FluentBit.java

```diff
@@ -0,0 +1,87 @@
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2019-2021 The Fluent Bit Authors
+ *  Copyright (C) 2015-2018 Treasure Data Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.fluentbit.test;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+
+import static io.fluentbit.test.Util.isPortOpen;
+import static io.fluentbit.test.Util.waitFor;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * This class wraps the Fluent-Bit process.
+ */
+public class FluentBit implements AutoCloseable {
+
+    private final Process fluentbit;
+
+    public FluentBit() {
+        URL config = FluentBit.class.getResource("/fluentbit.conf");
+        if (config == null) {
+            throw new RuntimeException("Can't find fluentbit.conf");
+        }
+
+        try {
+            ProcessBuilder processBuilder = new ProcessBuilder()
+                    .inheritIO()
+                    .command(
+                            Config.getFluentbitStartCommand()
+                                    .orElse(Arrays.asList(
+                                            "docker", "run",
+                                            "--rm",
+                                            "-e", "LOG_DEST_HOST",
+                                            "-e", "LOG_DEST_PORT",
+                                            "--network=host",
```

Comment:
We use the maven container to start the fluent bit container. The maven container needs to talk to fluent-bit and fluent-bit needs to talk to the maven container.

I have no idea how your CI/CD setup looks like. An alternative would be to create a docker network and then let both run in it. But that depends on your CI/CD setup.

Reply:
We use Github actions to do everything - have a look in the .github/workflows folder.

My main question here was does this require any fixed ports that may or may not be available or is it all ephemeral?

Reply:
This is for the case you want to locally run a docker image. In the CI/CD system I would grab from a previous build step a binary and mount it into the maven container. Of course the cpu arch needs to match.

Reply:
CPU arch should not need to match - for CI there are no ARM options for hosted Github runners for example so we use QEMU to emulate.

I think we need to integrate this with CI to confirm and use configurable ports wherever possible.
Again, my original question on why we need host networking is still not clear? It will cause issues with port conflicts so we should only expose the ports we need - how exactly do the Maven and Fluent Bit containers communicate, which ports? My preference would be a dedicated network for it, that seems a lot more resilient.